app/src/main/java/com/amaze/filemanager/filesystem/FileUtil.java

```diff
@@ -1072,5 +1072,29 @@ public OutputStream write(long size)
         }
 
     }
+    /**
+     * Returns the absolute FilePath of an URI
+     */
+    public static String getAbsoluteFilePath(Context c, Uri uri) {
+        String filePath = null;
+        Cursor cursor;
+        if (uri != null) {
+            if ("content".equals(uri.getScheme())) {
+                cursor = c.getContentResolver().query(uri, new String[]{MediaStore.Images.ImageColumns.DATA}, null, null, null);
```

Comment:
Have you tested this code with Google Drive and other third-party ContentProviders? AFAIK, using value of `_data` column like this does not really work with any `content://` Uris except ones from media storage (and media storage only works, because it's files are stored in public location on sdcard).